File: java/p1_TwoSum.java
/*
*    Problem: https://leetcode.com/problems/two-sum/description/
*    Title: Two Sum
*    Author: Kyungwoo Song
*    Date: 2018.02.17
*    Java Version : 1.8.0
*/

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> m = new HashMap();
		int n = nums.length;
		for(int i=0; i<n; i++)
		{
			int num = nums[i];
			m.put(num, i);
		}
		

		for(int i=0; i<n; i++)
		{
			int num = nums[i];
			int candidate = target - num;
			
			if ( (m.containsKey(candidate)) && (m.get(candidate) > i) )
			{
				return new int[] {i, m.get(candidate)};
			}
		}
		return null;
        
        
        
    }
}

File: python/p1_TwoSum.py
'''
*    Problem: https://leetcode.com/problems/two-sum/description/
*    Title: Two Sum
*    Author: Kyungwoo Song
*    Date: 2018.02.17
*    Python Version : 3.6
'''

class Solution:
    def twoSum(self, nums, target):
        num_itr_dict = dict()
        for itr in range(len(nums)):
            num = nums[itr]
            num_itr_dict[num] = itr

        for itr in range(len(nums)):
            num = nums[itr]
            candidate = target - num
            if (candidate in num_itr_dict) and (num_itr_dict[candidate]>itr):
                return [itr,num_itr_dict[candidate]]
